src/spac/spatial_analysis.py

```diff
@@ -86,19 +99,45 @@ def Cluster_Interaction_Matrix_Analysis(
 
         return ax
 
+    # Error Check Section
+    # -----------------------------------------------
     if not isinstance(adata, anndata.AnnData):
         error_text = "Input data is not an AnnData object. " + \
             f"Got {str(type(adata))}"
         raise ValueError(error_text)
 
-    # Extract column name
-    column_names = adata.obs.columns.tolist()
-    column_names_str = ",".join(column_names)
-
-    if annotation not in column_names:
-        error_text = f"Annotation {annotation} not found in the " + \
-            f"dataset. Existing annotations are: {column_names_str}"
-        raise ValueError(error_text)
+    # Check if annotation is in the dataset
+    check_annotation(
+        adata,
+        [annotation]
+    )
+
+    # Check if stratify_by is list or list of str
+    if stratify_by:
+        if not isinstance(stratify_by, str):
+            if isinstance(stratify_by, list):
+                for item in stratify_by:
+                    if not isinstance(item, str):
+                        error_text = "Item in the stratify_by " + \

```

Comment:
@ruiheesi, do we need that check given that the utility function check_annotation should take care of it.

Reply:
Good catch George, I will remove this.